src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/ReflectionJsonTypeInfoOfT.cs

```diff
@@ -28,20 +29,20 @@ internal ReflectionJsonTypeInfo(JsonConverter converter, JsonSerializerOptions o
             {
                 AddPropertiesAndParametersUsingReflection();
 
-#if NET7_0_OR_GREATER
                 // Compiler adds RequiredMemberAttribute to type if any of the members is marked with 'required' keyword.
                 // SetsRequiredMembersAttribute means that all required members are assigned by constructor and therefore there is no enforcement
                 bool shouldCheckMembersForRequiredMemberAttribute =
-                    typeof(T).IsDefined(typeof(RequiredMemberAttribute), inherit: true)
-                    && !(converter.ConstructorInfo?.IsDefined(typeof(SetsRequiredMembersAttribute), inherit: true) ?? false);
+                    typeof(T).HasRequiredMemberAttribute()
```

Comment:
I still think that this logic should be unified with the rest of the attribute mapping logic inside `AddPropertiesAndParametersUsingReflection`. I don't think it makes sense to re-walk the entire property cache simply to avoid doing if checks on `shouldCheckMembersForRequiredMemberAttribute` on construction of each property.

Reply:
shouldCheckMembersForRequiredMemberAttribute needed a bit plumbing but attribute check now happens in JsonPropertyInfo along with other properties